great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -963,9 +963,10 @@ def _validate_add_datasource_args(
         datasource: BaseDatasource | FluentDatasource | LegacyDatasource | None,
     ) -> None:
         if not ((datasource is None) ^ (name is None)):
-            raise ValueError(
-                "Must either pass in an existing datasource or individual constructor arguments (but not both)"
-            )
+            error_message = "Must either pass in an existing datasource or individual constructor arguments"
+            if datasource and name:
+                error_message += " (but not both)"
+            raise ValueError(error_message)
```

Comment:
I think this should be a `TypeError`
If you fail to provide a required positional argument you'll get a `TypeError`
```python
def foo(arg1, arg2):
    pass

foo()
... TypeError: foo() missing 2 required positional arguments: 'arg1' and 'arg2'
```